src/main/java/com/hivemq/mqtt/topic/tree/TopicTreeImpl.java

```diff
@@ -983,4 +990,40 @@ private boolean isEmptyArray(@Nullable final Object[] array) {
         return true;
     }
 
+    class FindSubscriberConsumer implements BiConsumer<SubscriberWithQoS, String> {
```

Comment:
static?

Reply:
The class uses the `createDistinctSubscribers()` method, which is not static and uses more non-static methods.
Therefore, I guess I can't make this one static, unfortunately...

Reply:
what about:
  final FindSubscriberConsumer subscriberConsumer = new FindSubscriberConsumer(client, this);
...
  public FindSubscriberConsumer(final @NotNull String client, final @NotNull TopicTreeImpl topicTree) {
            this.client = client;
            this.topicTree = topicTree;
        }
...
topicTree.createDistinctSubscribers(subscribers);